policy-format.md

```diff
@@ -0,0 +1,172 @@
+should be fatal ot nor?
+
+## Support Policy Items
+
+### Content Security Policy (CSP)
+
+The `content-security-policy` policy item contains the equivalent of one or more
+[Content-Security-Policy HTTP headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP).
+
+- The `"content-security-policy"` policy item contains an array of
+  content-security-policy directives.
+- Each CSP directive contains a `"policy"` string, containing the CSP format.
+- A `"disposition"` is either `"enforce"` or `"report"`.
+
+Example:
+
+```js
+"content-security-policy": [
+  {
+    "policy": "frame-ancestors 'none'; object-src 'none';",
+    "disposition": "enforce"
+  },{
+    "policy": "script-src 'self' https://cdn.example.com/js/",
+    "disposition": "report",
+  }
+]
+```
+
+
+### Transport-Level Security (TLS)
+
+The `tls` policy item contains several directives related to TLS, particularly:
+
+- [HTTP Strict Transport Security](https://tools.ietf.org/html/rfc6797)
+- [HTTP Public Key Pinning](https://tools.ietf.org/html/rfc7469)
+- [Expect-CT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect-CT)
+
+Example:
+
+```js
+"tls": {
+  // Strict-Transport-Security
+  "required": true,
+
+  // Public-Key-Pins / Public-Key-Pins-Report-Only
+  "pinned-public-keys": {
+    "pins": [
+      "d6qzRu9zOECb90Uez27xWltNsj0e1Md7GkYYkVoZWmM=",
+      "E9CZ9INDbd+2eRQozYqqbQ2yXLVKB9+xcprMF+44U1g=",
+      "LPJNul+wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQ="
+    ],
+    "disposition": "enforce",
+    "report-to": "group-name",
+  },
+
+  // Expect-CT
+  "certificate-transparency": {
+    "disposition": "enforce",    // (or "report-only")
+    "report-to": "group-name"
+  },
+
+  // Others. `Expect-Staple`, etc.
+  ...
+}
+```
+
+Note: The actual format does not include comment syntax.
+
+### Feature Policy
+
+The `features` policy item contains the equivalent of one or more
+[Feature-Policy](https://wicg.github.io/feature-policy/) HTTP headers.
+
+Example:
+
+```js
+"features": {
+  "policy": "geolocation 'self' https://example.com"
+}
+```
+
+### Referrer Policy
+
+Referrer policy might look like:
+
+Example:
+
+```js
+"referrer": {
+ "policy": "origin-when-cross-origin",
+}
+```
+
+### Client Hints
+
+Example:
+
+```js
+"client-hints": [ "DPR", "Width", "Viewport-Width" ],
+```
+
+### CORS:
+
+Example:
+
+```js
+"cors-preflights": {
+  "no-credentials": {
+    "origins": "*",
+  },
+  "unsafe-include-credentials": {
+    "origins": [ "https://trusted.example.com/" ]
+  },
+}
+```
+
+# Notes & Appendices
+
+## Open Questions
+
+- Are there any generic properties (like 'must understand') that apply to all
+  policy items, or is this merely a collection of otherwise unrelated policy
+  items.
+- Error handling: (E.g., reject malformed policies?) This would likely need
+  to be fully defined for cross-browser compatibility, but the exact definition
+  will have substantial impact on deployability and security. The present
+  course here seems unclear.
```

Comment:
We need a story here regardless: what happens if a server doesn't send JSON but CBOR or YAML because they're silly?

Reply:
See comment above. For a general format error, I think 'fatal error' is correct.

I'd imagine a policy directive for origin policy itself, where it could declare error handling to be "fatal", "report", "ignore", or something. That's give page authors the means to test out their policies, get assurance that the error rates are negligible, and only then go into full enforcement.